File: app/src/main/java/line/tree/TreeActivity.java
package line.tree;


import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;

import com.example.yingfu.line.R;

import line.util.ToolUtil;

public class TreeActivity extends AppCompatActivity implements View.OnClickListener {
    private Tree mTree;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.tree_activity);

        TreeNode treeNode3 = new TreeNode("3", null, null);
        TreeNode treeNode4 = new TreeNode("4", null, null);
        TreeNode treeNode5 = new TreeNode("5", null, null);
        TreeNode treeNode6 = new TreeNode("6", null, null);

        TreeNode treeNode1 = new TreeNode("1", treeNode3, treeNode4);
        TreeNode treeNode2 = new TreeNode("2", treeNode5, treeNode6);

        TreeNode root = new TreeNode("0", treeNode1, treeNode2);

        mTree = new Tree();
        mTree.setRoot(root);

        findViewById(R.id.level).setOnClickListener(this);
        findViewById(R.id.prePrint).setOnClickListener(this);
        findViewById(R.id.inPrint).setOnClickListener(this);
        findViewById(R.id.postPrint).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.level:
                ToolUtil.toast("level: " + mTree.getLevel());
                Log.e("ying>>>", "onClick: level: " + mTree.getLevel());
                break;
            case R.id.prePrint:
                ToolUtil.toast("prePrint: " + mTree.preTraversal());
                Log.e("ying>>>", "onClick: prePrint: " + mTree.preTraversal());
                break;
            case R.id.inPrint:
                ToolUtil.toast("inPrint: " + mTree.inTraversal());
                Log.e("ying>>>", "onClick: inPrint: " + mTree.inTraversal());
                break;
            case R.id.postPrint:
                ToolUtil.toast("postPrint: " + mTree.postTraversal());
                Log.e("ying>>>", "onClick: postPrint: " + mTree.postTraversal());
                break;

        }
    }
}

File: app/src/main/java/line/javafoundation/tree/TreeActivity.kt
package line.javafoundation.tree

import android.os.Bundle
import android.util.Log
import android.view.View
import com.example.yingfu.line.R
import line.BaseActivity
import line.javafoundation.file.file
import line.util.ToolUtil.toast

class TreeActivity : BaseActivity(), View.OnClickListener {
    private var mTree: Tree? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.tree_activity)
        val treeNode3 = TreeNode("3", null, null)
        val treeNode4 = TreeNode("4", null, null)
        val treeNode5 = TreeNode("5", null, null)
        val treeNode6 = TreeNode("6", null, null)
        val treeNode1 = TreeNode("1", treeNode3, treeNode4)
        val treeNode2 = TreeNode("2", treeNode5, treeNode6)
        val root = TreeNode("0", treeNode1, treeNode2)
        mTree = Tree()
        mTree!!.root = root
        findViewById<View>(R.id.level).setOnClickListener(this)
        findViewById<View>(R.id.prePrint).setOnClickListener(this)
        findViewById<View>(R.id.inPrint).setOnClickListener(this)
        findViewById<View>(R.id.postPrint).setOnClickListener(this)
        readFile()
    }

    private fun readFile() {
        val file = file()
        Log.e("ying>>>>1", file.readAssetsFile(this))
        file.writeFile(this)
        Log.e("ying>>>>22", file.readFile(this))
        file.writeFile(this)
        Log.e("ying>>>>333", file.readFile(this))
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.level -> {
                toast("level: " + mTree!!.level)
                Log.e("ying>>>", "onClick: level: " + mTree!!.level)
            }
            R.id.prePrint -> {
                toast("prePrint: " + mTree!!.preTraversal())
                Log.e("ying>>>", "onClick: prePrint: " + mTree!!.preTraversal())
            }
            R.id.inPrint -> {
                toast("inPrint: " + mTree!!.inTraversal())
                Log.e("ying>>>", "onClick: inPrint: " + mTree!!.inTraversal())
            }
            R.id.postPrint -> {
                toast("postPrint: " + mTree!!.postTraversal())
                Log.e("ying>>>", "onClick: postPrint: " + mTree!!.postTraversal())
            }
        }
    }
}